client/lib/cgutil/cgutil_linux.go

```diff
@@ -1,121 +1,106 @@
+//go:build linux
+
 package cgutil
 
 import (
+	"fmt"
 	"os"
 	"path/filepath"
 
-	cgroupFs "github.com/opencontainers/runc/libcontainer/cgroups/fs"
-
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/nomad/helper/uuid"
 	"github.com/opencontainers/runc/libcontainer/cgroups"
-	"github.com/opencontainers/runc/libcontainer/cgroups/fscommon"
-	"github.com/opencontainers/runc/libcontainer/configs"
-	"golang.org/x/sys/unix"
-)
-
-const (
-	DefaultCgroupParent      = "/nomad"
-	SharedCpusetCgroupName   = "shared"
-	ReservedCpusetCgroupName = "reserved"
+	lcc "github.com/opencontainers/runc/libcontainer/configs"
 )
 
-func GetCPUsFromCgroup(group string) ([]uint16, error) {
-	cgroupPath, err := getCgroupPathHelper("cpuset", group)
-	if err != nil {
-		return nil, err
-	}
+// UseV2 indicates whether only cgroups.v2 is enabled. If cgroups.v2 is not
+// enabled or is running in hybrid mode with cgroups.v1, Nomad will make use of
+// cgroups.v1
+//
+// This is a read-only value.
+var UseV2 = cgroups.IsCgroup2UnifiedMode()
+
+// GetCgroupParent returns the mount point under the root cgroup in which Nomad
+// will create cgroups. If parent is not set, an appropriate name for the version
+// of cgroups will be used.
+func GetCgroupParent(parent string) string {
+	if UseV2 {
+		return getParentV2(parent)
+	}
+	return getParentV1(parent)
+}
 
-	man := cgroupFs.NewManager(&configs.Cgroup{Path: group}, map[string]string{"cpuset": cgroupPath}, false)
-	stats, err := man.GetStats()
-	if err != nil {
-		return nil, err
+// CreateCPUSetManager creates a V1 or V2 CpusetManager depending on system configuration.
+func CreateCPUSetManager(parent string, logger hclog.Logger) CpusetManager {
+	if UseV2 {
+		return NewCpusetManagerV2(getParentV2(parent), logger.Named("cpuset.v2"))
 	}
-	return stats.CPUSetStats.CPUs, nil
+	return NewCpusetManagerV1(getParentV1(parent), logger.Named("cpuset.v1"))
 }
 
-func getCpusetSubsystemSettings(parent string) (cpus, mems string, err error) {
-	if cpus, err = fscommon.ReadFile(parent, "cpuset.cpus"); err != nil {
-		return
-	}
-	if mems, err = fscommon.ReadFile(parent, "cpuset.mems"); err != nil {
-		return
+// GetCPUsFromCgroup gets the effective cpuset value for the given cgroup.
+func GetCPUsFromCgroup(group string) ([]uint16, error) {
+	if UseV2 {
+		return getCPUsFromCgroupV2(getParentV2(group))
 	}
-	return cpus, mems, nil
+	return getCPUsFromCgroupV1(getParentV1(group))
 }
 
-// cpusetEnsureParent makes sure that the parent directories of current
-// are created and populated with the proper cpus and mems files copied
-// from their respective parent. It does that recursively, starting from
-// the top of the cpuset hierarchy (i.e. cpuset cgroup mount point).
-func cpusetEnsureParent(current string) error {
-	var st unix.Statfs_t
+// CgroupScope returns the name of the scope for Nomad's managed cgroups for
+// the given allocID and task.
+//
+// e.g. "<allocID>-<task>.scope"
```

Comment:
I think we want a dot to match what we've done in the code everywhere:


```suggestion
// e.g. "<allocID>.<task>.scope"
```